component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/services/TOTPAdminService.java

```diff
@@ -217,7 +219,59 @@ public String retrieveSecretKey(String username, AuthenticationContext context)
                     claims.put(TOTPAuthenticatorConstants.ENCODING_CLAIM_URL, encoding);
                     TOTPKeyGenerator.addTOTPClaimsAndRetrievingQRCodeURL(claims, username, context);
                 } else {
-                    secretKey = TOTPUtil.decrypt(secretKey);
+                    secretKey = TOTPUtil.decryptSecret(new String(secretKey));
+                }
+            }
+        } catch (AuthenticationFailedException e) {
```

Comment:
Why do we need to introduce these catch blocks for the change?

Reply:
These exceptions were already there. These are showing as newly added as we have deprecated retrieveSecretKey() method and introduced retrieveSecretKeyInByte().